spec/lib/efile/nj/nj1040_calculator_spec.rb

```diff
@@ -623,12 +623,58 @@ def over_65_birth_year
     end
   end
 
+  describe 'line 28b - other retirement income exclusion' do
+    let(:intake) { create(:state_file_nj_intake) }
+
+    context 'when taxpayer is ineligible' do
+      it 'sets line 28b to 0' do
+        allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return false
+        instance.calculate
+        expect(instance.lines[:NJ1040_LINE_28B].value).to eq(0)
+      end
+    end
+
+    context 'when taxpayer is eligible' do
+      context 'when maximum_exclusion minus 28a is greater than total_eligible_nonretirement_income' do
+        it 'sets line 28b to total_eligible_nonretirement_income' do
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return true
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:total_eligible_nonretirement_income).and_return 1_000
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:calculate_maximum_exclusion).and_return 50_000
+          allow(instance).to receive(:calculate_line_28a).and_return 0
+          instance.calculate
+          expect(instance.lines[:NJ1040_LINE_28B].value).to eq(1000)
+        end
+      end
+
+      context 'when maximum_exclusion minus 28a is less than total_eligible_nonretirement_income' do
+        it 'sets line 28b to maximum_exclusion minus 28a' do
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:eligible?).and_return true
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:total_eligible_nonretirement_income).and_return 100_000
+          allow_any_instance_of(Efile::Nj::NjRetirementIncomeHelper).to receive(:calculate_maximum_exclusion).and_return 50_000
+          allow(instance).to receive(:calculate_line_28a).and_return 0
+          instance.calculate
+          expect(instance.lines[:NJ1040_LINE_28B].value).to eq(50_000)
+        end
+      end
+    end
+  end
+
+  describe 'line 28c - total exclusion amount' do
+    it 'sets line 28c to 28a plus 28b' do
+      allow(instance).to receive(:calculate_line_28a).and_return 1_000
+      allow(instance).to receive(:calculate_line_28b).and_return 2_000
+      instance.calculate
+      expect(instance.lines[:NJ1040_LINE_28C].value).to eq(3_000)
+    end
+  end
+
   describe 'line 29 - gross income' do
     let(:intake) { create(:state_file_nj_intake) }
 
-    it 'sets line 29 to the sum of all state wage amounts' do
+    it 'sets line 29 to the sum of all state wage amounts minus 28c' do
       allow(instance).to receive(:calculate_line_15).and_return 50000
       allow(instance).to receive(:calculate_line_16a).and_return 1_000
+      allow(instance).to receive(:calculate_line_28c).and_return 0
```

Comment:
[pebble] same comment as above - by mocking this to zero, I don't think we're gaining any confidence that this is actually being used in the calculation. I think this test would still pass even if we removed the changes from the calculate_line_29 method. Can we set it to a number and change the expected output to match?